pages/content/amp-dev/documentation/guides-and-tutorials/optimize-measure/server-side-rendering.md

```diff
@@ -0,0 +1,74 @@
+
+In order to avoid content jumps, AMP requires websites to add the [AMP boilerplate code]({{g.doc('/content/amp-dev/documentation/guides-and-tutorials/learn/spec/amp-boilerplate.md', locale=doc.locale).url.path}}) in the header.  The AMP boilerplate renders the page invisible by setting the page body's opacity to 0.  Once AMP has been loaded, it is able to calculate the layout of the page. After that, AMP sets the body's opacity to 1 making the page content visible. 
+
+This way AMP is able to avoid content jumps during page load.  However, the downside of this approach is that the first render of the page doesn't happen until the AMP framework has been downloaded and executed.
+
+To improve this, AMP pages can be rendered server-side. This means AMP layouts such as the `responsive` or `fixed-height` layout are rendered on the server before serving the page to the user agent. This way it becomes possible to remove the AMP boilerplate **and** avoid content jumps during page load. 
+
+Important: AMP Caches already perform these optimizations. However, they're still important as AMP pages are not always served from an AMP Cache.
+
+## How does it work?
+
+The basic idea is to render the AMP layouts (`responsive`, `fixed-height`,....) server-side instead of client-side. Here is an example showing the differences between an AMP page and it's server-side rendered version ([click for a large version](/static/img/docs/guides/optimized-amp-diff.png)).
+
+<a href="/static/img/docs/guides/optimized-amp-diff.png"><amp-img lightbox layout="responsive" width="2560" height="773" src="/static/img/docs/guides/optimized-amp-diff.png"></amp-img></a> 
+
+Let's take a closer look at the individual changes:
+
+&#8291;**1. Mark the document as `transformed=self`:** Server-side rendered AMP pages are still valid AMP pages. However, the AMP validator needs to know that the AMP page contains server-side rendered AMP markup. This is done by adding the `transformed=self` attribute to the `<html>` element.
+
+```html
+<html ⚡ i-amphtml-layout i-amphtml-no-boilerplate transformed="self">
+```
+
+Note: **Note:** AMP caches use a different transformed flag, for example, the Google AMP caches adds `transformed=google;v1`.
+
+&#8291;**2. Remove the AMP boilerplate (when possible):** The [amp-boilerplate code](https://amp.dev/documentation/guides-and-tutorials/learn/spec/amp-boilerplate) has been removed. This means that page content is no longer hidden initially and renders straight away. This greatly increases the perceived performance of an AMP page. It's not always possible to remove the AMP boilerplate. For example, some AMP components, such as `amp-experiment` or `amp-dynamic-css-classes`, modify the page layout based on the user agent, which cannot be server-side rendered. In order to avoid content jumps, these need to be loaded before the page content can be displayed, which requires the AMP boilerplate code to initially hide the content of the page.
```

Comment:
Note: We should probably document this caveat on the reference docs for amp-experiment and amp-dynamic-css-classes.